gateway/api_definition.go

```diff
@@ -319,6 +326,16 @@ func (a APIDefinitionLoader) MakeSpec(def *apidef.APIDefinition, logger *logrus.
 
 	spec.APIDefinition = def
 
+	apiString, err := json.Marshal(def)
+	if err != nil {
+		logger.WithError(err).WithField("name", def.Name).Error("Failed to JSON marshal API definition")
+		return spec
+	}
+
+	sha256hash := sha256.Sum256(apiString)
```

Comment:
Rather than using this inline, maybe we could reuse `storage.HashStr`? The returned value is a hex string, which doesn't require urlencoding.